substrate.go

```diff
@@ -8,6 +8,8 @@ import (
 // Message is the single type that represents all messages in substrate.
 type Message interface {
 	Data() []byte
+	// Key is used as the partition key for messages
```

Comment:
A partition key is not defined in proximo and is a kafka specific concept.

Reply:
Yeah, I'm aware. This comment doesn't accurately reflect the idea behind it. What it more accurate is something along the lines of: 

```
Key provides a way to optionally specify a routing/partitioning key for backends which support this functionality.
```